Add output intermediate mesh option to MeshGenerator (#23486)
idaholab/moose: framework/doc/content/source/meshgenerators/MeshGenerator.md

GeneratedMeshGenerator 'gmg':    Bounding box delta: (x,y,z)=(       1,        1,        0)
GeneratedMeshGenerator 'gmg':   Global mesh volume = 1
```

> The parameter [!param](/Mesh/GeneratedMeshGenerator/output) output a mesh file immediately after it is generated. It can be used to view the intermediate mesh block during mesh generation.

Reviewer: ```suggestion
The parameter [!param](/Mesh/GeneratedMeshGenerator/output) enables the output of a mesh file immediately after it is generated. It can be used to view the intermediate meshes during mesh generation.
```

Take the simple example:

```
[Mesh]
  [./left]
    type = GeneratedMeshGenerator
    dim = 2
    nx = 3
    ny = 3
    xmin = -3
    xmax = 0
    ymin = -5
    ymax = 5
    output = true
    nemesis = true
  [../]
  [./right]
    type = GeneratedMeshGenerator
    dim = 2
    nx = 3
    ny = 3
    xmin = 3
    xmax = 6
    ymin = -5
    ymax = 5
  [../]

  [./left_and_right]
    type = MeshCollectionGenerator
    inputs = 'left right'

>   [../]

Reviewer: Let's include the input file from the test instead so that they're linked

[]
```

> The above will result in an intermediate mesh file 'left_in.e' besides the final mesh file 'output_intermediate_mesh_in.e'. This feature works for both REPLICATED and DISTRIBUTED mesh type.

Reviewer: ```suggestion
The above will result in an intermediate mesh file 'left_in.e' in addition to the final mesh file 'output_intermediate_mesh_in.e' when ran in mesh only mode.
```

We need to chat about the replicated/distributed case. I haven't decided if we want nemesis output by default. Some people will run in parallel, but then want to use the output mesh as a serialized mesh for input and then this won't work. I think we may need to provide the option to output distributed as exodus.
